doc/dev/transforms.rst

```diff
@@ -0,0 +1,594 @@
+Catalyst Compiler Passes
+########################
+
+..
+    TODO: add MLIR syntax highlighting to these snippets
+    TODO: add an end-to-end guide which includes compiling and using the custom pass
+
+External resources
```

Comment:
Maybe another title would be nice, more in relation with MLIR. Something like  `MLIR an transformations`

Reply:
Good suggestion, thanks!

Reply:
https://github.com/PennyLaneAI/catalyst/pull/340/commits/37ad9f76ea5a7c37099ef5a67b466cdf5faa84ae